RabbitMQClientLibrary/MessagePublishingService/RabbitMqMessagePublisher.cs

```diff
@@ -0,0 +1,61 @@
+﻿using Newtonsoft.Json;
+using RabbitMQ.Client;
+using RabbitMQClientLibrary.Configuration;
+using RabbitMQClientLibrary.MessagePublishingService;
+using System.Text;
+
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace ServerStatisticsCollection.Services
+{
+    public class RabbitMqMessagePublisher : IMessageQueueClientPublisher
+    {
+        private readonly RabbitMQConfiguration _configuration;
+        private readonly string _defaultExchangeName = "server_statistics_exchange";
+        private readonly ILogger<RabbitMqMessagePublisher> _logger;
+        public RabbitMqMessagePublisher(IOptions<RabbitMQConfiguration> rabbitMqConfig, ILogger<RabbitMqMessagePublisher> logger)
+        {
+            _configuration = rabbitMqConfig.Value;
+            _logger = logger;
+        }
+
+        public async Task PublishAsync<T>(string topic, T message, string? exchangeName = null)
+        {
+            try
+            {
+                using var connection = CreateConnection();
```

Comment:
can't we open the connection and channels in the costructor so we can reuse them rather than openning them eachtime we send a message